server/src/main/java/io/crate/auth/HostBasedAuthentication.java

```diff
@@ -122,6 +125,9 @@ public static HBAConf parse(Settings settings) {
 
     private final Supplier<String> clusterId;
 
+    // JwkProvider are cached to reused in JWTAuthenticationMethod and create on per-invocation base
+    private final ConcurrentHashMap<String, JwkProvider> cachedJwkProviders = new ConcurrentHashMap<>();
```

Comment:
Phrasing is a bit confusing - reads like we create JwkProvider per invocation base.

Maybe can be smth like
> JWT token is checked per connection, caching JwkProvider per issuer, so that each new JWTAuthenticationMethod can re-use cache.

Reply:
👍